### PR TITLE
Add headers that provide extra security protection in browsers

### PR DESCRIPTION
**Pull request by jwforres**

### CODE REVIEW

**Comment by jwforres** [comment]
@liggitt @spadgett PTAL

File: pkg/assets/handlers.go [hunk 1]
```diff
@@ -52,6 +52,15 @@ func GzipHandler(h http.Handler) http.Handler {
 	})
 }
 
+func BrowserSecurityHandler(h http.Handler) http.Handler {
```
Comment (spadgett) [inline pkg/assets/handlers.go]: I have a small preference for `SecurityHeadersHandler`, which makes it more obvious what the handler is adding

**Comment by jwforres** [comment]
yeah I need to run gofmt, forgot about that

**Comment by jwforres** [comment]
updated the handler name and ran gofmt, just need to rebuild to double check everything is still working

**Comment by jwforres** [comment]
ok verified this builds and is working, @liggitt can you take a quick glance and then i think it is good to merge

File: pkg/assets/handlers.go [hunk 2]
```diff
@@ -52,6 +52,15 @@ func GzipHandler(h http.Handler) http.Handler {
 	})
 }
 
+func SecurityHeadersHandler(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Add("X-Content-Type-Options", "nosniff")
+		w.Header().Add("X-XSS-Protection", "1; mode=block")
+		w.Header().Add("X-Frame-Options", "DENY")
```
Comment (liggitt) [inline pkg/assets/handlers.go]: `Add` or `Set`?

File: pkg/assets/mime.go [hunk 3]
```diff
@@ -24,12 +24,11 @@ func RegisterMimeTypes() {
...
 	// http://www.w3.org/TR/WOFF/#appendix-b
 	registerIfNeeded(".woff", "application/font-woff")
+	// https://www.w3.org/TR/WOFF2/#IMT
+	registerIfNeeded(".woff2", "application/font-woff2")
 	// http://www.iana.org/assignments/media-types/application/font-sfnt
 	registerIfNeeded(".ttf", "application/font-sfnt")
 	registerIfNeeded(".otf", "application/font-sfnt")
-
-	// Flash
-	registerIfNeeded(".swf", "application/x-shockwave-flash")
```
Comment (liggitt) [inline pkg/assets/mime.go]: leave this just in case an extension was including a swf

**Comment by liggitt** [comment]
a couple nits, LGTM

**Comment by jwforres** [comment]
i will switch these to Set since this is new code, and I highly doubt
anything else in our handler chain would be setting these, but we should
think about whether our other header Adds should be changed as a separate
issue

**Comment by jwforres** [comment]
updated, but i need to rebuild/retest in the morning

**Comment by jwforres** [comment]
ok all comments have been addressed [merge]

**Comment by jwforres** [comment]
fixed the extension unit test that was expecting text/javascript [merge]

**Pull request merged**